### PR TITLE
Fix logging for import and unmanage flows

### PR DESCRIPTION
**Pull request by anmolsachan**
tendrl-bug-id: Tendrl/commons#1009
bugzilla: 1590661
Signed-off-by: Anmol Sachan <anmol13694@gmail.com>

### CODE REVIEW

File: tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py [hunk 1]
```diff
@@ -10,6 +10,15 @@ def __init__(self, *args, **kwargs):
         super(DeleteClusterDetails, self).__init__(*args, **kwargs)
 
     def run(self):
+        logger.log(
+            "info",
+            NS.publisher_id,
+            {
+                "message": "Deleting cluster details."
```
Comment (shtripat) [inline tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py]: Doesn't the atom executor code says executing this atom already? Dow we need this log again?

File: tendrl/commons/objects/node/atoms/is_node_tendrl_managed/__init__.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 from tendrl.commons import objects
 from tendrl.commons.objects import AtomExecutionFailedError
```
Comment (shtripat) [inline tendrl/commons/objects/node/atoms/is_node_tendrl_managed/__init__.py]: Do you need this import anymore?

File: tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py (hunk 1 above)
Comment (anmolsachan) [reply tendrl/commons/objects/cluster/atoms/delete_cluster_details/__init__.py]: It is not giving for cluster details, that is why it needs to be added.

File: tendrl/commons/objects/node/atoms/is_node_tendrl_managed/__init__.py (hunk 2 above)
Comment (anmolsachan) [reply tendrl/commons/objects/node/atoms/is_node_tendrl_managed/__init__.py]: Will change this.

File: tendrl/commons/flows/__init__.py [hunk 3]
```diff
@@ -250,8 +252,13 @@ def _execute_atom(self, atom_fqdn):
                     parameters=self.parameters
                 ).run()
                 return ret_val
-            except AtomExecutionFailedError as ex:
-                raise ex
+            except AtomExecutionFailedError:
```
Comment (GowthamShanmugam) [inline tendrl/commons/flows/__init__.py]: Already traceback is identified here https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271

File: tendrl/commons/flows/import_cluster/__init__.py [hunk 4]
```diff
@@ -121,4 +125,8 @@ def run(self):
             if _errors:
                 _cluster.errors = _errors
             _cluster.save()
-            raise ex
+            raise FlowExecutionFailedError(str(
```
Comment (GowthamShanmugam) [inline tendrl/commons/flows/import_cluster/__init__.py]: same case here https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271

File: tendrl/commons/flows/unmanage_cluster/__init__.py [hunk 5]
```diff
@@ -88,4 +92,8 @@ def run(self):
             if _errors:
                 _cluster.errors = _errors
             _cluster.save()
-            raise ex
+            raise FlowExecutionFailedError(str(
```
Comment (GowthamShanmugam) [inline tendrl/commons/flows/unmanage_cluster/__init__.py]: same case no need traceback https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271

File: tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py [hunk 6]
```diff
@@ -41,6 +40,4 @@ def run(self):
                     flow_id=self.parameters['flow_id'],
                     integration_id=NS.tendrl_context.integration_id
                 )
-                raise AtomExecutionFailedError(
-                    "Cluster data sync still incomplete. Timing out"
-                )
+                return False
```
Comment (GowthamShanmugam) [inline tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py]: log the proper message before return False, just return False again leads to confusion. take the message which is present in AtomExecutionFailedError in old code and log it using logger before return False.

File: tendrl/commons/flows/unmanage_cluster/__init__.py [hunk 7]
```diff
@@ -70,6 +73,7 @@ def run(self):
         except (FlowExecutionFailedError,
                 AtomExecutionFailedError,
                 Exception) as ex:
+            exc_type, exc_value, exc_traceback = sys.exc_info()
```
Comment (GowthamShanmugam) [inline tendrl/commons/flows/unmanage_cluster/__init__.py]: no need to find traceback

File: tendrl/commons/flows/__init__.py (hunk 3 above)
Comment (anmolsachan) [reply tendrl/commons/flows/__init__.py]: The trace-back defined in https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271 gives trace-back till this point and not any further files in import flow. When we add trace-back capture here, it goes till the error where any of the AtomExecutionFailedError occur. Currently we have a few of them after this PR, But still this gives a better traceback.

File: tendrl/commons/objects/cluster/atoms/check_cluster_nodes_up/__init__.py [hunk 8]
```diff
@@ -69,6 +68,14 @@ def run(self):
             # no need to check for child job
             return all_node_status_up
         except (etcd.EtcdKeyNotFound, TypeError) as ex:
-            raise FlowExecutionFailedError(
-                "Error checking status of nodes .error: %s" % str(ex)
+            logger.log(
+                "error",
```
Comment (GowthamShanmugam) [inline tendrl/commons/objects/cluster/atoms/check_cluster_nodes_up/__init__.py]: this is correct do same thing in all place where you are doing return False, log the message and return False

File: tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py (hunk 6 above)
Comment (anmolsachan) [reply tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py]: It is present, It is present in file, Please try expanding the file

File: tendrl/commons/objects/cluster/atoms/check_cluster_nodes_up/__init__.py (hunk 8 above)
Comment (anmolsachan) [reply tendrl/commons/objects/cluster/atoms/check_cluster_nodes_up/__init__.py]: yes

File: tendrl/commons/flows/import_cluster/__init__.py (hunk 4 above)
Comment (anmolsachan) [reply tendrl/commons/flows/import_cluster/__init__.py]: The trace-back defined in https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271 gives trace-back till the flow file exception capture and not any further files in import flow. When we add trace-back capture here, it goes till the error where any of the AtomExecutionFailedError occur. Currently we have a few of them after this PR, But still this gives a better traceback.

File: tendrl/commons/flows/unmanage_cluster/__init__.py (hunk 5 above)
Comment (anmolsachan) [reply tendrl/commons/flows/unmanage_cluster/__init__.py]: The trace-back defined in https://github.com/Tendrl/commons/blob/master/tendrl/commons/jobs/__init__.py#L271 gives trace-back till the flow file exception capture and not any further files in import flow. When we add trace-back capture here, it goes till the error where any of the AtomExecutionFailedError occur. Currently we have a few of them after this PR, But still this gives a better traceback.

File: tendrl/commons/objects/node/flows/stop_services/__init__.py [hunk 9]
```diff
@@ -11,14 +11,24 @@ def run(self):
...
+            )
             srv = NS.tendrl.objects.Service(service=service)
             if not srv.running:
                 logger.log(
                     "debug",
                     NS.publisher_id,
                     {
                         "message": "%s not running on "
-                        "%s" % (service, NS.node_context.fqdn)
+                        "%s" % (service, NS.node_context.node_id)
```
Comment (GowthamShanmugam) [inline tendrl/commons/objects/node/flows/stop_services/__init__.py]: use fqdn in all unmanage log message and job updates, node_id is not readable by users. I saw all the place node_id is used in unmanage flows and atoms

File: tendrl/commons/objects/cluster/atoms/check_sync_done/__init__.py [hunk 10]
```diff
@@ -17,23 +16,24 @@ def run(self):
...
+                node_id=NS.node_context.node_id
+            ).load()
+            if _cnc.first_sync_done is not None and \
                     _cnc.first_sync_done.lower() == "yes":
-                    break
-            except etcd.EtcdKeyNotFound:
-                loop_count += 1
-                continue
+                break
+            loop_count += 1
```
Comment (GowthamShanmugam) [inline tendrl/commons/objects/cluster/atoms/check_sync_done/__init__.py]: tested?

**Comment by GowthamShanmugam** [comment]
Some minor comments only, it looks fine. Please verify in all the cases

File: tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py (hunk 6 above)
Comment (GowthamShanmugam) [reply tendrl/commons/objects/cluster/atoms/check_cluster_available/__init__.py]: looks good

File: tendrl/commons/objects/node/flows/stop_services/__init__.py (hunk 9 above)
Comment (GowthamShanmugam) [reply tendrl/commons/objects/node/flows/stop_services/__init__.py]: import cluster flow is using fqdn is everywhere better we use fqdn in unmanage flow also

Comment (anmolsachan) [reply tendrl/commons/objects/node/flows/stop_services/__init__.py]: In unmanage cluster, most of the places node_id is used. Stop services comes under unmanage. That is why node_id is used to maintain uniformity in unmanage flow.

Comment (GowthamShanmugam) [reply tendrl/commons/objects/node/flows/stop_services/__init__.py]: i feel we need to change all places in unmanage flow. @shtripat

File: tendrl/commons/objects/cluster/atoms/check_sync_done/__init__.py (hunk 10 above)
Comment (anmolsachan) [reply tendrl/commons/objects/cluster/atoms/check_sync_done/__init__.py]: Yes, this is tested

**Comment by anmolsachan** [comment]
@shtripat @r0h4n Please review.

File: tendrl/commons/objects/node/flows/stop_services/__init__.py (hunk 9 above)
Comment (r0h4n) [reply tendrl/commons/objects/node/flows/stop_services/__init__.py]: Please use fqdn only.

**Comment by anmolsachan** [comment]
@r0h4n In Unmanage flow, we are using node_id in all the messages, stop-services is a part of that flow. Thus I have used node_id there. If we have to use fqdn, we have to add logic to find fqdn for the unmanage flow, because it is executed from the tendrl server, and only stop services runs on nodes, that's why fqdn will be present there but not in other parts of Unmanage atoms.

**Comment by anmolsachan** [comment]
@r0h4n Please review.

**Comment by r0h4n** [comment]
Please fix unit tests @anmolsachan

**Comment by anmolsachan** [comment]
@r0h4n I have already fixed tests related to this pr. The tests are broken are not related to this PR, thats why I did not include them in this.

**Comment by r0h4n** [comment]
@anmolsachan Cant merge any PR with failing unit tests now

You can fix the unit tests in a new PR (or take help from the interns) and rebase this current one on top of it

**Comment by anmolsachan** [comment]
@r0h4n Will Do that.

**Comment by anmolsachan** [comment]
@r0h4n This is good to be merged now. My bad, a unit test had failed due to recent changes.

**Comment by anmolsachan** [comment]
@r0h4n This is good to go. Please review.

**Pull request merged**